pkg/registry/apis/provisioning/register.go

```diff
@@ -175,15 +176,110 @@ func RegisterAPIService(
-				// for now????
+		func(ctx context.Context, a authorizer.Attributes) (decision authorizer.Decision, reason string, err error) {
+			if identity.IsServiceIdentity(ctx) {
+				// A Grafana sub-system should have full access. We trust them to make wise decisions.
 				return authorizer.DecisionAllow, "", nil
 			}
 
-			// fallback to the standard authorizer
-			return authorizer.DecisionNoOpinion, "", nil
+			// Different routes may need different permissions.
+			// * Reading and modifying a repository's configuration requires administrator privileges.
+			// * Reading a repository's limited configuration (/stats & /settings) requires viewer privileges.
+			// * Reading a repository's files requires viewer privileges.
+			// * Editing a repository's files requires editor privileges.
+			// * Syncing a repository requires editor privileges.
+			// * Exporting a repository requires administrator privileges.
+			// * Migrating a repository requires administrator privileges.
+			// * Testing a repository configuration requires administrator privileges.
+			// * Viewing a repository's history requires editor privileges.
+
+			id, err := identity.GetRequester(ctx)
+			if err != nil {
+				return authorizer.DecisionDeny, "failed to find requester", err
+			}
+
+			switch a.GetResource() {
+			case provisioning.RepositoryResourceInfo.GetName():
+				// TODO: Support more fine-grained permissions than the basic roles. Especially on Enterprise.
+				switch a.GetSubresource() {
+				case "":
+					// Doing something with the repository itself.
+					if id.GetOrgRole().Includes(identity.RoleAdmin) {
+						return authorizer.DecisionAllow, "", nil
+					} else {
+						return authorizer.DecisionDeny, "admin role is required", nil
+					}
+
+				case "test", "export", "migrate":
+					// Testing doesn't make sense for non-admins.
+					// Exporting and migrating are potentially dangerous.
+					if id.GetOrgRole().Includes(identity.RoleAdmin) {
+						return authorizer.DecisionAllow, "", nil
+					} else {
+						return authorizer.DecisionDeny, "admin role is required", nil
+					}
+
+				case "webhook":
+					// When the resource is a webhook, we'll deal with permissions manually by checking signatures or similar in the webhook handler.
+					// The user in this context is usually an anonymous user, but may also be an authenticated synthetic check by the Grafana instance's operator as well.
+					// For context on the anonymous user, check the authn/clients/provisioning.go file.
+					return authorizer.DecisionAllow, "", nil
+
+				case "files":
+					// Reading files is allowed for everyone, so as to allow code reviews and similar.
+					// Writing files is only allowed fo Editor and higher.
+					isViewer := id.GetOrgRole().Includes(identity.RoleViewer)
+					isEditor := id.GetOrgRole().Includes(identity.RoleEditor)
+					isReadOperation := a.GetVerb() == apiutils.VerbGet || a.GetVerb() == apiutils.VerbList || a.GetVerb() == apiutils.VerbWatch
```

Comment:
```suggestion
					isReadOperation := a.GetVerb() == apiutils.VerbGet || a.GetVerb()
```
list and watch verbs do not exist for sub-resource

Reply:
TIL, thanks!